Metal/Stores/HCIRNFilterStore.swift

```diff
@@ -0,0 +1,68 @@
+
+//    Copyright 2005-2021 Michel Fortin
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+import Foundation
+import CoreImage
+
+public class HCIRNFilterStore: FilterStore {
+
+    public private(set) var visionFilter: CIFilter? = nil
+    public private(set) var visionSimulation: NSInteger = 1
+
+    init(vision: NSInteger) {
+        HCIRNFilterVendor.registerFilters()
+        setVisionFilter(to: vision)
+    }
+}
+
+public extension HCIRNFilterStore {
+
+    func setVisionFilter(to vision: NSInteger) {
+        self.visionSimulation = vision
+        DispatchQueue.global().async { [weak self] in
+            self?.visionFilter = self?.loadFilter(for: vision)
+        }
```

Comment:
Is this thread-safe? You are assigning to `visionFilter` from a global queue here while the same `visionFilter` variable is accessed by `applyFilter` that gets called from the queue in `CGWindowListScreenCapturer`.

Reply:
Correct. CIFilters are not thread safe. Yet, there are no user-adjusted dynamic values, so it shouldn't rear its head. But, yeah, it's absolutely incorrect. I'll change how these filters are handled to make it clean and address another concern re: letting windows have different active filters.